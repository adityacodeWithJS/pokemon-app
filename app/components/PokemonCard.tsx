import Image from "next/image";
import Link from "next/link";
type Pokemeon = {
  id: number;
  name: string;
  imageUrl: string;
}


const pokemonCard = ({ id, name, imageUrl }: Pokemeon) => {
  return (
    <Link href={`/pokemon/${id}`}>
      <div className="bg-white shadow-lg rounded-lg transform transition duration-300 hover:scale-105">
        <Image
          src={imageUrl}
          width={200}
          height={200}
          alt="loading"
          className=" h-32 object-contain"
        />
        <div className="p-4 font-semibold text-gray-800">
          <h2>{name}</h2>
        </div>
      </div>
    </Link>

  )
}

export default pokemonCard