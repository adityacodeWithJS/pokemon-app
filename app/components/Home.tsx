import {  PokemeonCard } from "../interface/types";
import PokemonCard from "./PokemonCard"
import React from "react";

const Home = ({pokemonList}:any) => {
    
return <>
<div className='grid grid-cols-1 sm:grid-cols-4  gap-3 m-5'>
        {
          pokemonList.length > 0 &&
          pokemonList.map((pokemon: PokemeonCard, index: number) => {
            return (
              <PokemonCard key={index + '_002'} {...pokemon} />
            )
          })
        }
       
      </div>
</>
}
export default Home