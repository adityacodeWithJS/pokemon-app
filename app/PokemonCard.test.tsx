import React from 'react'
import { render, screen } from '@testing-library/react';
import Home from './components/Home';
import'@testing-library/jest-dom';

import  PokemonCard from "./components/PokemonCard"
const pokemonListMock=[
    {
      "name": "bulbasaur",
      "url": "https://pokeapi.co/api/v2/pokemon/1/"
    },
    {
      "name": "ivysaur",
      "url": "https://pokeapi.co/api/v2/pokemon/2/"
    },
    {
      "name": "venusaur",
      "url": "https://pokeapi.co/api/v2/pokemon/3/"
    },
    {
      "name": "charmander",
      "url": "https://pokeapi.co/api/v2/pokemon/4/"
    },
    {
      "name": "charmeleon",
      "url": "https://pokeapi.co/api/v2/pokemon/5/"
    },
    {
      "name": "charizard",
      "url": "https://pokeapi.co/api/v2/pokemon/6/"
    },
    {
      "name": "squirtle",
      "url": "https://pokeapi.co/api/v2/pokemon/7/"
    },
    {
      "name": "wartortle",
      "url": "https://pokeapi.co/api/v2/pokemon/8/"
    },
    {
      "name": "blastoise",
      "url": "https://pokeapi.co/api/v2/pokemon/9/"
    },
    {
      "name": "caterpie",
      "url": "https://pokeapi.co/api/v2/pokemon/10/"
    }
  ]
describe('PokemonCard',() =>{
    const mockPokemon = {
        id:  4,
        imageUrl: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png",
        name: "charmander"
    }

    test('renderAllCards',() => {
        render (<PokemonCard {...mockPokemon}/>)
    })

    test('renderHomePage',() => {
        render (<Home  pokemonList={pokemonListMock}/>)
    })
})

