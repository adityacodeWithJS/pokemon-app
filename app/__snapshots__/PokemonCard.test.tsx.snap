// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard renderAllCards 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/pokemon/4"
      >
        <div
          class="bg-white shadow-lg rounded-lg transform transition duration-300 hover:scale-105"
        >
          <img
            alt="loading"
            class=" h-32 object-contain"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F4.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F4.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F4.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div
            class="p-4 font-semibold text-gray-800"
          >
            <h2>
              charmander
            </h2>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/pokemon/4"
    >
      <div
        class="bg-white shadow-lg rounded-lg transform transition duration-300 hover:scale-105"
      >
        <img
          alt="loading"
          class=" h-32 object-contain"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F4.png&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F4.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F4.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
        <div
          class="p-4 font-semibold text-gray-800"
        >
          <h2>
            charmander
          </h2>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
